src/honey/sql.cljc

```diff
@@ -2283,50 +2292,38 @@
+         numbered? (:numbered opts @default-numbered)
+         formatter (if (map? data) #'format-dsl #'format-expr)
+         options {:caching cache
+                  :checking (:checking opts @default-checking)
+                  :clause-order (if dialect?
+                                  (if-let [f (:clause-order-fn dialect)]
+                                    (f @base-clause-order)
+                                    @current-clause-order)
                                   @current-clause-order)
-                                @current-clause-order)
-               *ignored-metadata* (if (contains? opts :ignored-metadata)
-                                    (:ignored-metadata opts)
-                                    [])
-               *inline*  (cond (contains? opts :inline)
-                               (:inline opts)
-                               (= :nrql (:dialect dialect))
-                               true
-                               :else
-                               @default-inline)
-               *numbered* (when numbered
-                            (atom []))
-               *quoted*  (cond (contains? opts :quoted)
-                               (:quoted opts)
-                               (= :nrql (:dialect dialect))
-                               nil
-                               dialect?
-                               true
-                               :else
-                               @default-quoted)
-               *quoted-always* (if (contains? opts :quoted-always)
-                                 (:quoted-always opts)
-                                 @default-quoted-always)
-               *quoted-snake* (if (contains? opts :quoted-snake)
-                                (:quoted-snake opts)
-                                @default-quoted-snake)
-               *params* (reduce-kv (fn [m k v]
-                                     (assoc m (sym->kw k) v))
-                                   {}
-                                   (:params opts))
-               *values-default-columns* (:values-default-columns opts)]
+                  :ignored-metadata (:ignored-metadata opts [])
+                  :inline (:inline opts (if (= :nrql (:dialect dialect))
+                                          true
+                                          @default-inline))
+                  :numbered (when numbered? (atom []))
+                  :quoted (cond (contains? opts :quoted)
+                                (:quoted opts)
+                                (= :nrql (:dialect dialect))
+                                nil
+                                dialect?
+                                true
+                                :else
+                                @default-quoted)
+                  :quoted-always (:quoted-always opts @default-quoted-always)
+                  :quoted-snake (:quoted-snake opts @default-quoted-snake)
+                  :params (not-empty
```

Comment:
Why the addition of `not-empty` here?

Reply:
It was relevant for the previous approach; can be removed now.